logic/src/commonTest/kotlin/com/wire/kalium/logic/data/featureConfig/FeatureConfigRepositoryTest.kt

```diff
@@ -0,0 +1,107 @@
+package com.wire.kalium.logic.data.featureConfig
+
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.test_util.TestNetworkException
+import com.wire.kalium.logic.util.shouldFail
+import com.wire.kalium.logic.util.shouldSucceed
+import com.wire.kalium.network.api.featureConfigs.FeatureConfigApi
+import com.wire.kalium.network.api.featureConfigs.FeatureConfigResponse
+import com.wire.kalium.network.exceptions.KaliumException
+import com.wire.kalium.network.utils.NetworkResponse
+import io.mockative.Mock
+import io.mockative.classOf
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.once
+import io.mockative.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class FeatureConfigRepositoryTest {
+
+    @Test
+    fun whenFileSharingFeatureConfigSuccess_thenTheSuccessIsReturned() = runTest {
+        // Given
+        val expectedSuccess = Either.Right(FileSharingModel(lockStatus = "locked", status = "enabled"))
+        val (arrangement, featureConfigRepository) = Arrangement().arrange()
+        arrangement.withSuccessfulResponse()
+
+        // When
+        val result = featureConfigRepository.getFileSharingFeatureConfig()
+
+        // Then
+        result.shouldSucceed { expectedSuccess.value }
+        verify(arrangement.featureConfigApi)
+            .suspendFunction(arrangement.featureConfigApi::fileSharingFeatureConfig)
+            .wasInvoked(once)
+    }
+
+    @Test
+    fun whenFileSharingFeatureConfigFailWithOperationDeniedError_thenTheErrorIsPropagated() = runTest {
+        // Given
+        val operationDeniedException = TestNetworkException.operationDenied
+        val (arrangement, featureConfigRepository) = Arrangement().arrange()
+        arrangement.withErrorResponse(operationDeniedException)
```

Comment:
ideally `arrange()` should be called last after all the settings options have been set/provided.